lib/ui/src/modules/ui/components/layout/index.js

```diff
@@ -157,18 +149,22 @@ class Layout extends React.Component {
   }
 
   onResize(pane, mode) {
-    return size => {
-      this.layerSizes[pane][mode] = size;
-      saveSizes(this.layerSizes);
-
-      const { clientWidth, clientHeight } = this.previewPanelRef;
+    let resizeTimeout;
 
-      this.setState({
-        previewPanelDimensions: {
-          width: clientWidth,
-          height: clientHeight,
-        },
-      });
+    return size => {
+      clearTimeout(resizeTimeout);
+      resizeTimeout = setTimeout(() => {
+        this.layerSizes[pane][mode] = size;
+        saveSizes(this.layerSizes);
+
+        const { clientWidth, clientHeight } = this.previewPanelRef;
+        this.setState({
+          previewPanelDimensions: {
+            width: clientWidth,
+            height: clientHeight,
+          },
+        });
+      }, 50);
```

Comment:
Please use `lodash.debounce` package

Reply:
Or maybe it should rather be `lodash.throttle`, see https://github.com/storybooks/storybook/pull/3180#issuecomment-372009581

Reply:
@Hypnosphi Sounds good. I discovered a new issue with using `_.throttle` the way that it is currently implemented. it will create a new throttled function on every state update which is bad. Thus I will be changing how the `onResize` function and how its passed in to the `onChange` handlers

Reply:
Just apply it in constructor

Reply:
I tend to use that method of writing bound class methods a lot.

Reply:
for readability purposes, it might be better to name it `this.onThrottledResize` to differentiate between the two.